File: src/telegram.rs
const ERROR_REQUEST_FAILED: &str = "request to Telegram failed";

pub fn send_message(token: String, channel: String, message: String) {
    let endpoint = format!("https://api.telegram.org/bot{}/sendMessage", token);

    ureq::post(endpoint.as_str())
        .send_form(&[
            ("chat_id", channel.as_str()),
            ("text", message.as_str()),
            ("parse_mode", "MarkdownV2"),
        ])
        .expect(ERROR_REQUEST_FAILED);
}

File: telegram.go
package main

import (
	"fmt"
	"log"
	"net/http"
	"net/url"
)

const ErrorRequestFailed = "request to Telegram failed"

func sendMessage(token, channel, message string) {
	endpoint := fmt.Sprintf("https://api.telegram.org/bot%s/sendMessage", token)
	form := url.Values{
		"chat_id":    {channel},
		"text":       {message},
		"parse_mode": {"MarkdownV2"},
	}

	_, err := http.PostForm(endpoint, form)
	if err != nil {
		log.Fatalln(ErrorRequestFailed)
	}
}
